internal/handlers/tigris/msg_find.go

```diff
@@ -73,18 +74,37 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
-	firstBatch := types.MakeArray(len(resDocs))
-	for _, doc := range resDocs {
-		firstBatch.Append(doc)
+	batchSize := len(resDocs)
+	if len(resDocs) > int(params.BatchSize) {
+		batchSize = int(params.BatchSize)
+	}
+
+	firstBatch := types.MakeArray(batchSize)
+	moreResults := types.MakeArray(0)
+
+	for i := 0; i < len(resDocs); i++ {
+		if i < batchSize {
+			firstBatch.Append(resDocs[i])
+		} else {
+			moreResults.Append(resDocs[i])
+		}
+	}
```

Comment:
Maybe something like this (to have fewer cyclomatic complexity and fewer operations):

```suggestion
	for i := 0; i < batchSize; i++ {
		firstBatch.Append(resDocs[i])
	}
	
	for i := batchSize; i < len(resDocs); i++ {
		moreResults.Append(resDocs[i])
	}
```

Also I suggest moving the common logic from pg and tigris to common